src/odemis/gui/comp/viewport.py

```diff
@@ -442,45 +442,29 @@ def _checkMergeSliderDisplay(self):
+            all_opt = all(isinstance(s, (FluoStream, StaticFluoStream, CLStream)) for s in streams)
 
             # If all images are optical, assume they are merged using screen blending and no
             # merge ratio is required
             if all_opt:
                 self.ShowMergeSlider(False)
             else:
-                # TODO: How is the order guaranteed? (Left vs Right)
-                # => it should be done in the MicroscopeView when adding a stream
-                # For now, special hack for the MicroscopeCanvas which always sets
-                # the EM image as "right" (ie, it's drawn last).
-                # If there is EM and Spectrum or CL, the spectrum/CL image is always
-                # set as "right" (ie, it's drawn last).
+                # TODO: For now the order is set in the MicroscopeCanvas, but it
+                # should be done in the MicroscopeView when adding a stream.
+                # The order is: (left) CL/Fluo < EM < Spectrum (right)
                 # Note: in practice, there is no AR spatial stream, so it's
                 # never mixed with any other stream.
-                if (
-                        any(isinstance(s, EMStream) for s in streams) and
-                        any(isinstance(s, (FluoStream, StaticFluoStream)) for s in streams)
-                ):
-                    # TODO: don't hard-code MD_AT_FLUO and MD_AT_EM but use the
-                    # acquisitionType of the corresponding streams.
-                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_FLUO)
-                    self.bottom_legend.set_stream_type(wx.RIGHT, model.MD_AT_EM)
-                elif (
-                        any(isinstance(s, EMStream) for s in streams) and
-                        any(isinstance(s, SpectrumStream) for s in streams)
-                ):
-                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_EM)
-                    self.bottom_legend.set_stream_type(wx.RIGHT, model.MD_AT_SPECTRUM)
-                elif (
-                        any(isinstance(s, EMStream) for s in streams) and
-                        any(isinstance(s, CLStream) for s in streams)
-                ):
-                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_EM)
-                    self.bottom_legend.set_stream_type(wx.RIGHT, model.MD_AT_CL)
-                else:
-                    self.bottom_legend.set_stream_type(wx.LEFT, streams[0].acquisitionType.value)
-                    self.bottom_legend.set_stream_type(wx.RIGHT, streams[1].acquisitionType.value)
-
+                def get_stream_prio(s):
+                    if isinstance(s, (FluoStream, StaticFluoStream, CLStream)):
+                        return 0
+                    if isinstance(s, EMStream):
+                        return 1
+                    if isinstance(s, SpectrumStream):
+                        return 2
```

Comment:
Why are multiple if statements preferred here over elif statements?

Reply:
No specific reason. Technically, this works, as there are "returns", but `elif` are more explicit. Moreover, there should have been an `else`, to handle the case the stream is nothing defined.
I've updated the commit to do this properly.